### ISSUE TITLE
CA1401 reports on public method in private nested class

### ISSUE THREAD

**Issue by bkoelman** [dotnet/roslyn-analyzers#792]
CA1401 (P/Invoke method should not be visible) reports on a public method that is declared in a private nested class. 

Because such a method is effectively invisible from outside the class, I believe this case should not be reported.

Repro example (v1.1.0):

```csharp
public sealed class TimerFontContainer
{
    private static class NativeMethods
    {
        [DllImport("gdi32.dll")]
        public static extern IntPtr AddFontMemResourceEx(IntPtr pbFont, uint cbFont, IntPtr pdv,
            [In] ref uint pcFonts);
    }
}
```
